nimble/host/mesh/pkg.yml

```diff
@@ -45,5 +45,5 @@ pkg.req_apis:
     - stats
 
 pkg.init:
-    bt_mesh_register_gatt: 500
-    ble_mesh_shell_init: 1000
+    bt_mesh_register_gatt: 'MYNEWT_VAL(BLE_MESH_SYSINIT_STAGE)'
+    ble_mesh_shell_init: 'MYNEWT_VAL(BLE_MESH_SYSINIT_STAGE_2)'
```

Comment:
Can we use BLE_MESH_SHELL_SYSINIT_STAGE instead of BLE_MESH_SYSINIT_STAGE_2?

Reply:
Thanks, @rymanluk.  I agree that `BLE_MESH_SYSINIT_STAGE_2` is a poor name.

I renamed it to `BLE_MESH_SYSINIT_STAGE_SHELL` rather than your suggestion of `BLE_MESH_SHELL_SYSINIT_STAGE`.  I don't have a preference either way, but I already chose this arbitrary convention for other packages with multiple sysinit stages.

I pushed this commit separately with the intent to fixup before merging.